### PR TITLE
[kube-prometheus-stack] Add an option to enable/disable `kubernetes-system-controller-manager` PrometheusRule

### PR DESCRIPTION
**Pull request by Dima-Diachenko**
Hi all!

#### What this PR does / why we need it
You forgot to add an option to _enable/disable_ `kubernetes-system-controller-manager` PrometheusRule.

Look, all files in the [rules folder](https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack/templates/prometheus/rules-1.14) contain option to disable or enable PrometheusRule deployment by set in the variables as `.Values.defaultRules.rules.<name>`, for example (scroll to the end of the line below):

https://github.com/prometheus-community/helm-charts/blob/a3a8328e9e6449c1fa02302715ecabb10663b275/charts/kube-prometheus-stack/templates/prometheus/rules-1.14/alertmanager.rules.yaml#L7

or

https://github.com/prometheus-community/helm-charts/blob/a3a8328e9e6449c1fa02302715ecabb10663b275/charts/kube-prometheus-stack/templates/prometheus/rules-1.14/config-reloaders.yaml#L7

But (I don't know why) such option is missed for [kubernetes-system-controller-manager](https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-prometheus-stack/templates/prometheus/rules-1.14/kubernetes-system-controller-manager.yaml):
https://github.com/prometheus-community/helm-charts/blob/a3a8328e9e6449c1fa02302715ecabb10663b275/charts/kube-prometheus-stack/templates/prometheus/rules-1.14/kubernetes-system-controller-manager.yaml#L7
In this file it depend only on `.Values.defaultRules.create` and `.Values.kubeControllerManager.enabled` but nothing from `.Values.defaultRules.rules` object. 

And I faced with the issue when I set false for all elements in `.Values.defaultRules.rules` object but I still saw 1 deployed PrometheusRule:
```
defaultRules:
    create: true
    rules:
      alertmanager: false
      etcd: false
      configReloaders: false
      general: false
      k8s: false
      kubeApiserverAvailability: false
      kubeApiserverBurnrate: false
      kubeApiserverHistogram: false
      kubeApiserverSlos: false
      kubelet: false
      kubeProxy: false
      kubePrometheusGeneral: false
      kubePrometheusNodeRecording: false
      kubernetesApps: false
      kubernetesResources: false
      kubernetesStorage: false
      kubernetesSystem: false
      kubeScheduler: false
      kubeStateMetrics: false
      network: false
      node: false
      nodeExporterAlerting: false
      nodeExporterRecording: false
      prometheus: false
      prometheusOperator: false
```

So I went to your chart and found minor misconfiguration. By my PR I will add *kubeControllerManager: true* option.

#### Which issue this PR fixes

I didn't create an issue but I can if you will want it. 

#### Special notes for your reviewer
I have tested Helm template locally - everything is OK.

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

Thanks!

### CODE REVIEW

**Review by monotek — CHANGES_REQUESTED** [review]
The rules are generated by the scripts in the hackmdirectory. 

Therefore you have to adjust the scripts or your changes get lost the next time they are run...

File: charts/kube-prometheus-stack/hack/sync_prometheus_rules.py [hunk 1]
```diff
@@ -90,7 +90,7 @@ def new_representer(dumper, data):
     'kubernetes-system-kube-proxy': ' .Values.kubeProxy.enabled .Values.defaultRules.rules.kubeProxy',
     'kubernetes-system-apiserver': ' .Values.defaultRules.rules.kubernetesSystem', # kubernetes-system was split into more groups in 1.14, one of them is kubernetes-system-apiserver
     'kubernetes-system-kubelet': ' .Values.defaultRules.rules.kubernetesSystem', # kubernetes-system was split into more groups in 1.14, one of them is kubernetes-system-kubelet
-    'kubernetes-system-controller-manager': ' .Values.kubeControllerManager.enabled',
+    'kubernetes-system-controller-manager': ' .Values.kubeControllerManager.enabled .Values.defaultRules.rules.kubeControllerManager',
```
Comment (Dima-Diachenko) [inline charts/kube-prometheus-stack/hack/sync_prometheus_rules.py:93]: Thanks, @monotek, I didn't know about it.
THese changes should be enough I think. Please review.

File: charts/kube-prometheus-stack/Chart.yaml [hunk 2]
```diff
@@ -22,7 +22,7 @@ name: kube-prometheus-stack
 sources:
   - https://github.com/prometheus-community/helm-charts
   - https://github.com/prometheus-operator/kube-prometheus
-version: 39.4.0
+version: 39.4.1
```
Comment (monotek) [inline charts/kube-prometheus-stack/Chart.yaml:25]: New feature should raise minor version.

Please also sign your commits.

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/kube-prometheus-stack/Chart.yaml (hunk 2 above)
Comment (Dima-Diachenko) [reply charts/kube-prometheus-stack/Chart.yaml:25]: I forgot to sign off the last commit. Fixed. 
What do you mean _raise the minor version_ - I bumped it. And it's not a new feature, it's a kind of bugfix.

Comment (monotek) [inline charts/kube-prometheus-stack/Chart.yaml:25]:
```suggestion
version: 39.5.0
```

**Review by monotek — CHANGES_REQUESTED** [review]

**Review by monotek — APPROVED** [review]

**Pull request merged**